### PR TITLE
Market field addition

### PR DESCRIPTION
**Pull request by mbifulco**
this also includes the migration necessary.

@tkeemon - take a good look at this and talk to me before you merge.  I haven't gotten it working locally, and I'm not sure why.

### CODE REVIEW

**Comment by tkeemon** [comment]
I'm having no trouble running: 
```
python manage.py lms --settings=devstack migrate --delete-ghost-migrations
python manage.py cms --settings=devstack migrate --delete-ghost-migrations
```

and seeing the market field in mysql:
```
mysql> describe auth_userprofile;
+--------------------+--------------+------+-----+---------+----------------+
| Field              | Type         | Null | Key | Default | Extra          |
+--------------------+--------------+------+-----+---------+----------------+
| id                 | int(11)      | NO   | PRI | NULL    | auto_increment |
| user_id            | int(11)      | NO   | UNI | NULL    |                |
| name               | varchar(255) | NO   | MUL | NULL    |                |
| language           | varchar(255) | NO   | MUL | NULL    |                |
| location           | varchar(255) | NO   | MUL | NULL    |                |
| meta               | longtext     | NO   |     | NULL    |                |
| courseware         | varchar(255) | NO   |     | NULL    |                |
| gender             | varchar(6)   | YES  | MUL | NULL    |                |
| mailing_address    | longtext     | YES  |     | NULL    |                |
| year_of_birth      | int(11)      | YES  | MUL | NULL    |                |
| level_of_education | varchar(6)   | YES  | MUL | NULL    |                |
| goals              | longtext     | YES  |     | NULL    |                |
| allow_certificate  | tinyint(1)   | NO   |     | NULL    |                |
| country            | varchar(2)   | YES  |     | NULL    |                |
| city               | longtext     | YES  |     | NULL    |                |
| organization       | longtext     | YES  |     | NULL    |                |
| market             | int(11)      | YES  |     | NULL    |                |
+--------------------+--------------+------+-----+---------+----------------+
17 rows in set (0.00 sec)
```

but it doesn't look like the field is being populated in mysql after registration for some reason:
```
mysql> select * from auth_userprofile where user_id=20;
+----+---------+------+----------+----------+------+------------+--------+-----------------+---------------+--------------------+------------------+-------------------+---------+------+--------------+--------+
| id | user_id | name | language | location | meta | courseware | gender | mailing_address | year_of_birth | level_of_education | goals            | allow_certificate | country | city | organization | market |
+----+---------+------+----------+----------+------+------------+--------+-----------------+---------------+--------------------+------------------+-------------------+---------+------+--------------+--------+
| 17 |      20 | tjk  |          |          |      | course.xml | NULL   | NULL            |          NULL | none               | testing devstack |                 1 |         | NULL | NULL         |   NULL |
+----+---------+------+----------+----------+------+------------+--------+-----------------+---------------+--------------------+------------------+-------------------+---------+------+--------------+--------+
1 row in set (0.00 sec)
```

I'm still looking into it.

File: lms/templates/register.html [hunk 1]
```diff
@@ -244,6 +244,22 @@ <h3 class="message-title">${_("Welcome {username}").format(username=extauth_id)}
         <h2 class="sr">${_("Additional Personal Information")}</h2>
 
         <ol class="list-input">
+
+          % if settings.REGISTRATION_EXTRA_FIELDS['market'] != 'hidden':
+          <li class="field group">
+            <div class="field ${settings.REGISTRATION_EXTRA_FIELDS['market']} select" id="field-market">
+              <label for="market">${_('Market')}</label>
+              <selecti id="market" name="market" ${'required aria-required="true"' if settings.REGISTRATION_EXTRA_FIELDS['market'] == 'required' else ''}>
```
Comment (tkeemon) [inline lms/templates/register.html]: Typo I needed to fix to get the dropdown to display correctly.

**Comment by tkeemon** [comment]
Ok, I figured it out: 
https://github.com/appsembler/edx-platform/commit/09b6c4d82edeb7b17f641382cef9874cf4d3219c

Make these two changes and the PR should work. Give it a shot locally and let me know if you're still having trouble getting it to work under your devstack.

**Comment by mbifulco** [comment]
Ah man, that change in views.py is tiny but very important, eh?  Testing this now.

**Comment by natea** [comment]
looks like it was just a typo? <selecti ...> should have been <select ...>?

**Comment by mbifulco** [comment]
Well, that too - but we weren't connecting the python view to the model for the market field, either.  @tkeemon  There shouldn't be any need to rerun migrations to test this, right? I'm actually still not seeing the market field populated in my DB locally.

**Comment by tkeemon** [comment]
@natea that was preventing the market dropdown from displaying correctly. The other change in the student/views.py file makes sure that the value is saved into into the db.

**Comment by tkeemon** [comment]
@mbifulco if you're not seeing the column in the mysql consle after running:
```
use edxapp
describe auth_userprofile;
```

Then I would try rerunning the manage.py commands from my first comment.

**Comment by mbifulco** [comment]
The column is there, I was misreading the output from my select statement.  Looks like we're good to go!

**Comment by tkeemon** [comment]
Done. I'll push to your server now.

**Comment by mbifulco** [comment]
At long last! Whew!

**Pull request merged**